### ISSUE TITLE
[7.7.0] Missing breaking change: null values for minimum_should_match

### ISSUE THREAD

**Issue by jayaddison** [elastic/elasticsearch#56812]
**Elasticsearch version** (`bin/elasticsearch --version`): `7.7.0`

**Plugins installed**: []

**JVM version** (`java -version`): `openjdk 14 2020-03-17`

**OS version** (`uname -a` if on a Unix-like system): `Linux basecamp 5.4.0-29-generic #33-Ubuntu SMP Wed Apr 29 14:32:27 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

**Description of the problem including expected versus actual behavior**:
During execution of a `bool` query with `minimum_should_match` value set to `null`, an `x_content_parse_exception` was raised:

```
[bool] failed to parse field [minimum_should_match]
```

**Steps to reproduce**:

 1. Perform a bool query against an ES 7.7.0 cluster:
```sh
curl -H 'Content-Type: application/json' localhost:9200/_search --data '{"query": {"bool": {"minimum_should_match": null}}}'
```
 2. Encounter the error response:
```
{"error":{"root_cause":[{"type":"x_content_parse_exception","reason":"[1:45] [bool] failed to parse field [minimum_should_match]"}],"type":"x_content_parse_exception","reason":"[1:45] [bool] failed to parse field [minimum_should_match]","caused_by":{"type":"illegal_state_exception","reason":"Can't get text on a VALUE_NULL at 1:45"}},"status":400}
```

**Expected behaviour**:
This query was valid in ES 7.6.2, and doesn't appear in the 'breaking changes' for 7.7.0 ([ref](https://www.elastic.co/guide/en/elasticsearch/reference/7.7/release-notes-7.7.0.html#breaking-7.7.0)), so this failure was unexpected.

 1. Perform a bool query against an ES 7.6.2 cluster:
```sh
curl -H 'Content-Type: application/json' localhost:9200/_search --data '{"query": {"bool": {"minimum_should_match": null}}}'
```
2. Encounter a valid search response:
```
{"took":0,"timed_out":false,"_shards":{"total":0,"successful":0,"skipped":0,"failed":0},"hits":{"total":{"value":0,"relation":"eq"},"max_score":0.0,"hits":[]}}
```

**Comment by cbuescher** [comment]
@jayaddison thanks for raising this, this is likely an unintended side effect of https://github.com/elastic/elasticsearch/pull/52880. If we accepted `null` here and treated it like the default we should continue to do so on at least 7.x versions.

**Comment by jayaddison** [comment]
Brilliant, and thanks for responding to this so quickly, @cbuescher.  I've been able to roll-forward my application's behaviour to cope with the change in null handling, but I imagine others may still appreciate the fix.

**Comment by bcbee** [comment]
Filed ES support case 00549044 for similar issue with `must`/`should`/`must_not`/`filter`. Very easy fix in our app but broke some stuff in production.

**Comment by lanerjo** [comment]
How is this closed? This issue is still present in 7.8.1

**Comment by cbuescher** [comment]
> This issue is still present in 7.8.1

I just checked this on a local test instance and on cloud in 7.8.1 and the reported issue is fixed there. Can you make sure that you are not running any older nodes in your cluster? If you are still having issues I'd like to point you to to the support forums over at https://discuss.elastic.co to do further digging since I cannot reproduce this issue here. Many thanks.
